library/index.c

```diff
@@ -530,10 +530,14 @@ int eblob_generate_sorted_index(struct eblob_backend *b, struct eblob_base_ctl *
 	struct eblob_map_fd src, dst;
 	int fd, err, len;
 	char *file, *dst_file;
+	int cache_is_empty = 0;
 
 	if (b == NULL || bctl == NULL)
 		return -EINVAL;
 
+	/* check that eblob cache is empty. If cache is empty we can skip iterating and flushing it */
+	cache_is_empty = eblob_cache_empty(b);
+
```

Comment:
*I do not see any risk of data corruption from these changes.*

But apparently data corruption is there, and there is no data corruption in 0.22.13 version, which was simultaneously used in parallel groups.

Please, remind me, what was the benefit of those changes in the first place? How faster did became eblob load time?